src/transformers/models/bert/modeling_bert.py

```diff
@@ -1517,13 +1518,29 @@ def forward(
 
         loss = None
         if labels is not None:
-            if self.num_labels == 1:
-                #  We are doing regression
-                loss_fct = MSELoss()
-                loss = loss_fct(logits.view(-1), labels.view(-1))
+            if self.problem_type is not None:
+                if self.problem_type == "single_column_regression":
+                    loss_fct = MSELoss()
+                    loss = loss_fct(logits.view(-1), labels.view(-1))
+                elif self.problem_type == "multi_column_regression":
+                    loss_fct = MSELoss()
+                    loss = loss_fct(logits.view(-1, self.num_labels), labels)
```

Comment:
I would group those into just `regression`. `loss = loss_fct(logits.view(-1, self.num_labels), labels)` will work in both cases and we can find out if really need to if there is one or several labels by looking at `config.num_labels.`

Reply:
fixed